sec_certs_page/common/bindings.py

```diff
@@ -0,0 +1,510 @@
+import os
+import json
+import datetime
+import asyncio
+import hashlib
+import shutil
+import urllib
+from typing import Tuple, Union, Any
+import requests
+import jwt
+import click
+from flask import current_app
+from pymongo.collection import Collection
+
+
+from .. import mongo
+
+Document = Any
+
+
+class Binding():
+    """ A binding data class
+    """
+
+    def __init__(self, cert_id: str, header_name: str):
+        """Binding object constructor
+
+        Args:
+            cert_id (str): certificate id
+            header_name (str): header name
+        """
+        self.cert_id: Union[str, int] = cert_id if not cert_id.isdigit() else int(
+            cert_id)
+        self.header_name: str = header_name
+        self.header_data: list[object] = []
+
+    def match_name(self, name: str) -> bool:
+        """ Function to match a string to the header_name
+
+        Args:
+            name (str): Match candidate
+
+        Returns:
+            bool: Whether the name matches to the binding obj header_name
+        """
+        return name == urllib.parse.quote(
+            self.header_name) or name == self.header_name
+
+    def update_headers_if_match(self, name: str, header: list[object]) -> None:
+        """ Update header_data if this objects header_name matches with given name
+
+        Args:
+            name (str): Match candidate
+            header (list[object]): New header data
+        """
+        if self.match_name(name):
+            self.header_data = header
+
+    def __str__(self) -> str:
+        return self.__repr__()
+
+    def __repr__(self) -> str:
+        return f"Binding {self.cert_id} to {self.header_name}"
+
+    def __getitem__(self, key: str) -> Union[str, int, list[object]]:
+        return {
+            "cert_id": self.cert_id,
+            "header_name": self.header_name,
+            "header_data": self.header_data
+        }[key]
+
+
+def url_to_local_path(url: str) -> str:
+    """ Transorms local URL to local path
+
+    Args:
+        url (str): URL to transform
+
+    Returns:
+        str: Transformed URL
+    """
+    parsed_url = urllib.parse.urlparse(url)
+    path = urllib.parse.unquote(parsed_url.path.strip("/"))
+    return path
+
+
+def github_url_to_api(url: str) -> str:
+    """ Transform the standard public github URL to a github API URL
+
+    Args:
+        url (str): URL to transform
+
+    Returns:
+        str: Transformed API URL
+    """
+    parsed_url = urllib.parse.urlparse(url)
+    api_url = f"https://api.github.com/repos{parsed_url.path}"
+    if parsed_url.query:
+        api_url += f"?{parsed_url.query}"
+    return api_url
+
+
+def get_filename_from_url(url: str) -> str:
+    """ Parse out the filename from a given url
+
+    Args:
+        url (str): URL to parse
+
+    Returns:
+        str: Parsed out filename
+    """
+    path_components = url.split("/")
```

Comment:
What about using `urlparse` here and only splitting its path?

```suggestion
    path_components = urllib.parse.urlparse(url).path.split("/")
```